sympy/solvers/tests/test_solveset.py

```diff
@@ -1898,19 +1898,43 @@ def test_solve_expo():
 
 
 # logarithmic tests
-@XFAIL
 def test_logarithmic():
     assert solveset_real(log(x - 3) + log(x + 3), x) == FiniteSet(
-        sqrt(10))
+        -sqrt(10), sqrt(10))
     assert solveset_real(log(x + 1) - log(2*x - 1), x) == FiniteSet(2)
-    assert solveset_real(log(x + 3) + log(1 + 3/x) - 3) == FiniteSet(
+    assert solveset_real(log(x + 3) + log(1 + 3/x) - 3, x) == FiniteSet(
         -3 + sqrt(-12 + exp(3))*exp(S(3)/2)/2 + exp(3)/2,
         -sqrt(-12 + exp(3))*exp(S(3)/2)/2 - 3 + exp(3)/2)
 
     eq = z - log(x) + log(y/(x*(-1 + y**2/x**2)))
     assert solveset_real(eq, x) == \
-        FiniteSet(-sqrt(y*(y - exp(z))), sqrt(y*(y - exp(z))))
+        Intersection(S.Reals, FiniteSet(-sqrt(y**2 - y*exp(z)),
+            sqrt(y**2 - y*exp(z)))) - \
+        Intersection(S.Reals, FiniteSet(-sqrt(y**2), sqrt(y**2)))
+    assert solveset_real(
+        log(3*x) - log(-x + 1) - log(4*x + 1), x) == FiniteSet(-S(1)/2, S(1)/2)
+    assert solveset(log(x**y) - y*log(x), x, S.Reals) == S.Reals
+
+@XFAIL
+def test_uselogcombine_2():
     eq = log(exp(2*x) + 1) + log(-tanh(x) + 1) - log(2)
     assert solveset_real(eq, x) == EmptySet()
+    eq = log(8*x) - log(sqrt(x) + 1) - 2
+    assert solveset_real(eq, x) == EmptySet()
+
+
+def test_is_logarithmic():
+    assert _is_logarithmic(log(x), x) is True
+    assert _is_logarithmic(log(x)*log(y), x) is True
+    assert _is_logarithmic(log(x)**2, x) is True
+    assert _is_logarithmic(log(x**y) - y*log(x), x) is True
```

Comment:
here, logcombine for `a*log(x) == log(x**a)` is only valid "if x is positive and a is real"...so if these conditions are not met, the solution will not be valid.

Reply:
Yeah the assumptions need to be taken care of, but it would still be a logarithmic equation irrespective of the assumption of the `free_symbols`

Reply:
It depends on your definition of logarithmic, right? If being logarithmic means being able to rewrite as a single log then assumptions matter. (That's why I said that a good definition is needed in the docstring.) I don't see a point in saying something is logarithmic if it can't be rearranged into a form suitable for solving...so that's my bias in terms of what the routine should say. But if you are going to always use force and return conditions for the coefficients and arguments of log then I guess the current beahvior is ok.

Reply:
I will be preferring this definition:

> Logarithmic equation is an equation that involves the logarithm of an expression containing a variable.

Is it fine?

Reply:
It's too vague...`sin(log(x)) = 0` is not logarithmic. You should be answering True if the equation is likely to benefit from rewriting via `logcombine`. That an equation might be logarithmic *at some point of inversion* doesn't mean that it is always logarithmic. Another example is `log(x)**2 - 2*log(x) - 1` which is polynomial with `log(x)` the generator.

Reply:
>  You should be answering True if the equation is likely to benefit from rewriting via logcombine

I see your point, but using `logcombine` can give sometime wrong answer. See for eg: `log(x) + log(2/x)`, reduces to `log(2)`, if I simply check that the equation gets reduced then the routine will return `True` for cases involving constants, like `log(2) + log(3)` will also give `True`, so to avoid this if I add a check to see `symbol` in `.args.free_symbols` then the first equation would return `False`, because it reduces to a constant.

>  Another example is log(x)**2 - 2*log(x) - 1 which is polynomial with log(x) the generator

For this example, correct me if I interpret logarithmic equation wrong, even though it is polynomial in nature it still can be said as logarithmic since `log` has `x` in its arguments. Also once we get the solutions of the polynomial `t**2 - 2*t - 1` we need to solve `log(x) - t` and this is again logarithmic equation.

Reply:
> Will your code allow this to pass into lambert testing or will it, after getting a ConditionSet solution from _solve_log, return as unsolved?

In this branch, it would return a `ConditionSet`, but as soon as I add lambert solving (in another PR) the equations that both `_solve_log` and `_solve_expo` can't solve will be passed into lambert solving. I don't think we will be having identification for lambert because they don't take any specific form, so we will directly solve those equations considering only possible way left.

> x*log(x) - 3 will be logarithmic with your definition but is actually a Lambert form

Yeah you are right but similar thing would happen with `_is_exponential()` too. For `exp(x) + x` the routine would return `True` but it is actually a lambert form.

Reply:
I think the definitions should be more precisely tuned so you don't have this ambiguity. 

Regarding Lambert forms...they don't have a *single* specific form but I recall going through some pains to elaborate and identify such equations in whatever form they came in the work that I did with `solve`.